src/views/figure_element.ts

```diff
@@ -83,6 +83,15 @@ export class FigureElementView extends BlockView<Trackable<FigureElementNode>> {
     let handleUpload = () => {
       //noop
     }
+    const handleNewFile = (): boolean => {
+      let figureUploaded = true
+      this.node.content.forEach((file) => {
+        if (file.attrs.src) {
+          figureUploaded = false
+        }
+      })
+      return figureUploaded
+    }
```

Comment:
I don't get that, is it a reference?

Reply:
This function checks if an image is uploaded inside a figure and disables the attachment options accordingly. When the image is removed, the options are re-enabled. The function works in most cases, but QA found an issue when the dropdown remains open(https://jira.atypon.com/secure/attachment/1850159/1850159_LEAN-2165-+retest+failed.webm) —the state is updated after opening the attachment icon, causing the logic to fail. I believe I might be missing something here or that the approach may need to be adjusted.

The goal is to disable the dropdown options when an image is uploaded and re-enable them once the image is detached or if no image is uploaded. Do you have any suggestions or ideas for a better way to handle this?